pkg/util/test/util.go

```diff
@@ -475,14 +455,25 @@ func validateStorageClusterPods(
 				cluster.Namespace, cluster.Name, err)
 		}
 
-		if len(pods) != expectedPodCount {
-			return "", true, fmt.Errorf("expected pods: %v. actual pods: %v", expectedPodCount, len(pods))
+		if len(pods) != len(expectedPxNodeNameList) {
+			return "", true, fmt.Errorf("expected pods: %v. actual pods: %v", len(expectedPxNodeNameList), len(pods))
 		}
 
+		var pxNodeNameList []string
+		var podsNotReady []string
 		for _, pod := range pods {
 			if !coreops.Instance().IsPodReady(pod) {
-				return "", true, fmt.Errorf("pod %v/%v is not yet ready", pod.Namespace, pod.Name)
+				podsNotReady = append(podsNotReady, pod.Name)
 			}
+			pxNodeNameList = append(pxNodeNameList, pod.Spec.NodeName)
+		}
+
+		if len(podsNotReady) > 0 {
+			return "", true, fmt.Errorf("pods are not Ready: %v", podsNotReady)
+		}
+
+		if !assert.ElementsMatch(&testing.T{}, expectedPxNodeNameList, pxNodeNameList) {
+			return "", false, fmt.Errorf("expected Portworx nodes: %+v, got %+v", expectedPxNodeNameList, pxNodeNameList)
```

Comment:
This should be `return "", true, ...` , right? We do want to retry if there are not enough pods, correct?

Reply:
No, we do not want to retry here, if those node arrays do not match, it's game over.